imgix_test.go

```diff
@@ -1,119 +1,3 @@
+// TODO: Discuss. I'm thinking we do integration testing here against
+// an actual source to ensure paths resolve correctly.
```

Comment:
There's a couple of things I've done in the past that could be used here

## 1. Writing an integration test that doesn't run by default
If you want the option to run an integration test against a real source, but don't want to run it under normal circumstances, you could

### a) Create a test that [skips](https://golang.org/pkg/testing/#hdr-Skipping) when the `go test -short` flag is used:
```go
func TestLiveRequest(t *testing.T) {
    if testing.Short() {
        t.Skip("skipping due to -short")
    }
}
```

But you'd have to run `go test -short` every time you didn't want it to run

### b) Create a test file that has a [build constraint](https://golang.org/cmd/go/#hdr-Build_constraints) that prevents it from running during normal test runs

The beginning of the test file would look like:
```go
// +build integration

package imgix
```

Then you could target the test like this:
```
go test -v -tags "integration" github.com/imgix/imgix-go
```

And that would run all the tests that had the build constraint `integration`. Without the tag specified, the test wouldn't be included in the run. I tend to do this for benchmarks.

## 2. Mock the server you want to issue a request to

If you're using the golang standard library to help mock requests, you can:

- Create a [handler function for an endpoint](https://github.com/imgix/prometheus-am-executor/blob/master/main.go#L267)
- [Call that endpoint directly](https://github.com/imgix/prometheus-am-executor/blob/master/main_test.go#L459) from a test case. This would give you the HTTP status code feedback you were looking for.

Depending on how your http server code was organized, you could also run an instance of your server in a test case and issue requests against it. The prometheus-am-executor code is [setup to](https://github.com/imgix/prometheus-am-executor/blob/master/main.go#L375) [do this](https://github.com/imgix/prometheus-am-executor/blob/master/main.go#L428), but so far issuing a request object to the handler directly has been sufficient.

Reply:
Archiving this for future reference @sherwinski